_source/_posts/2021-08-06-webforms-blazor-migration.md

```diff
@@ -0,0 +1,839 @@
+---
+layout: blog_post
+title: Web Forms Migration to Blazor in .NET Core
+author: giorgi-dalakishvili
+by: contractor
+communities: [.net]
+description: "Moving from .NET 4x Web Forms to Blazor? Here is a quick guide for that."
+tags: [blazor, webforms, migration, dotnet, dotnetcore, csharp, aspnet, aspnetcore, dotnet5]
+tweets:
+- "Learn how migrate to #Blazor from #WebForms in a snap ->"
+- "Get up to speed on moving to #Blazor from #WebForms in #dotnet"
+- "Wondering how #Blazor stacks up against #WebForms? We've got you covered in this migration tutorial!"
+image: blog/featured/okta-dotnet-blazor-closeup.jpg
+type: conversion
+---
+
+ASP.NET Web Forms framework has been the cornerstone technology of .Net for web development since the release of .Net Framework in 2002. ASP.NET Web Forms includes a layer of abstraction for developers so that you don't need to care about HTML, JavaScript, or any other front-end technology. It provides a development flow similar to building desktop apps, a way for developers to build a web page by drag and drop, an event-driven programming model, and stateful controls thanks to the built-in state management. These features incurred lots of complexity, but it was worthwhile when web development was still in its infancy.
+
+Web development has evolved a lot in recent years. Browsers have become more powerful and implement various standards that allow access to user's devices. Client-side frameworks such as React and Angular simplify client-side development and have become increasingly popular. These frameworks and other benefits are available only if you can write the front-end JavaScript code.
```

Comment:
```suggestion
Web development has evolved a lot in recent years. Browsers have become more powerful and implement various standards that allow access to users' devices. Client-side frameworks such as React and Angular simplify client-side development and have become increasingly popular. These frameworks and other benefits are available only if you can write the front-end JavaScript code.
```
grammar fix: it's either "a user's devices" OR "users' devices" - i went for the plural